From 1c3fc25af2b6ccaefaf205a72b5289347632fd91 Mon Sep 17 00:00:00 2001
From: Chunjian zheng <chunjian.zheng@amlogic.com>
Date: Fri, 12 Dec 2014 10:52:20 +0800
Subject: [PATCH 5201/5965] pd#83628: fix PMU repaid and trickle time bug.     
      do not update battery staus if it is mbox project.

Change-Id: I7237e27cfbbc480c80d9fad49a0c3149abdf91b3
---
 drivers/amlogic/power/aml_pmu/aml1218-sply.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/amlogic/power/aml_pmu/aml1218-sply.c b/drivers/amlogic/power/aml_pmu/aml1218-sply.c
index a3a2ecdc4d31..0466823aaf5f 100755
--- a/drivers/amlogic/power/aml_pmu/aml1218-sply.c
+++ b/drivers/amlogic/power/aml_pmu/aml1218-sply.c
@@ -426,7 +426,7 @@ int aml1218_set_trickle_time(int minutes)
 {
     int bits;
 
-    if (minutes < 30 && minutes > 80) {
+    if ( (minutes < 30) || (minutes > 80) ) {
         AML1218_ERR("%s, invalid trickle time:%d\n", __func__, minutes);
         return -EINVAL;
     }
@@ -445,7 +445,7 @@ int aml1218_set_rapid_time(int minutes)
 {
     int bits;
 
-    if (minutes > 360 || minutes < 720) {
+    if ( (minutes < 360) || (minutes > 720) ) {
         AML1218_ERR("%s, invalid rapid time:%d\n", __func__, minutes);
         return -EINVAL;
     }
@@ -1417,6 +1417,8 @@ static void aml1218_charging_monitor(struct work_struct *work)
         api->pmu_update_battery_capacity(charger, aml1218_battery); 
     } else {
         aml1218_update_state(charger);
+        schedule_delayed_work(&supply->work, supply->interval);
+        return;
     }
 
     /*
-- 
2.19.0

